### PR TITLE
conf: New classes for append options

### PR DESCRIPTION
**Pull request by m-blaha**
Introducing `OptionStringAppendList` and `OptionStringAppendSet` classes
for append options.

The current implementation of append options is flawed because it requires
the options to be set in ascending order of priority for proper 
functionality. This is necessary for the resetting mechanism using an empty
item to work correctly. However, in dnf5, options are not processed
strictly in priority order. For instance, command line options like
`--exclude` are processed before the configuration files, despite having a
higher priority than config files.

The new classes accumulate all setting attempts and only process them in 
priority order when the option value is needed.

Resolves: https://github.com/rpm-software-management/dnf5/issues/1331

### CODE REVIEW

**Review by jan-kolarik — COMMENTED** [review]
It might be useful to also add some more unit / CI tests for the empty values inside of the option containers to make sure it behaves like it should...

File: include/libdnf5/conf/option_string_list.hpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ namespace libdnf5 {
 /// Option that stores a container of strings. The type of the container is a template parameter.
 /// Support default value, and check of an input value using the regular expression
 // @replaces libdnf:conf/OptionStringList.hpp:class:OptionStringList
-template <typename T>
+template <typename T, bool APPEND = false>
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply include/libdnf5/conf/option_string_list.hpp:34]: Yeah, probably... I was kind of torn between considering it a constant or a variable. I'll change it.

File: libdnf5/conf/option_string_list.cpp [hunk 2]
```diff
@@ -53,55 +65,97 @@ class OptionStringContainer<T>::Impl {
...
+            if (values.empty()) {
+                // setting to empty value clears the result
+                retval.clear();
+                continue;
+            }
+            bool first = true;
+            for (const auto & item : values) {
+                if (item.empty()) {
+                    // if the first item in the list is empty, clear the result
+                    if (first) {
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply libdnf5/conf/option_string_list.cpp:95]: Oh, thanks! It got lost during copy&paste...

**Comment by m-blaha** [comment]
> It might be useful to also add some more unit / CI tests for the empty values inside of the option containers to make sure it behaves like it should...

Good point. I will add more tests.

**Review by jan-kolarik — APPROVED** [review]
